cartographer_ros/cartographer_ros/offline_node.cc

```diff
@@ -60,19 +64,39 @@ constexpr char kTfStaticTopic[] = "/tf_static";
 constexpr char kTfTopic[] = "tf";
 constexpr double kClockPublishFrequencySec = 1. / 30.;
 constexpr int kSingleThreaded = 1;
+// We publish tf messages one second earlier than other messages. Under
+// the assumption of higher frequency tf this should ensure that tf can
+// always interpolate.
+const ::ros::Duration kDelay = ::ros::Duration(1.0);
 
 void RunOfflineNode(const MapBuilderFactory& map_builder_factory) {
   CHECK(!FLAGS_configuration_directory.empty())
       << "-configuration_directory is missing.";
-  CHECK(!FLAGS_configuration_basename.empty())
-      << "-configuration_basename is missing.";
+  CHECK(!FLAGS_configuration_basenames.empty())
+      << "-configuration_basenames is missing.";
   CHECK(!FLAGS_bag_filenames.empty()) << "-bag_filenames is missing.";
   const auto bag_filenames =
       cartographer_ros::SplitString(FLAGS_bag_filenames, ',');
   cartographer_ros::NodeOptions node_options;
-  cartographer_ros::TrajectoryOptions trajectory_options;
-  std::tie(node_options, trajectory_options) = cartographer_ros::LoadOptions(
-      FLAGS_configuration_directory, FLAGS_configuration_basename);
+  const auto configuration_basenames =
+      cartographer_ros::SplitString(FLAGS_configuration_basenames, ',');
+  TrajectoryOptions first_trajectory_options;
+  std::tie(node_options, first_trajectory_options) =
+      LoadOptions(FLAGS_configuration_directory, configuration_basenames.at(0));
+
+  std::vector<TrajectoryOptions> trajectory_options;
```

Comment:
Remove first_trajectory_options.
Don't make == 1 a too special case.
Loop over trajectory_options with an index until bag_filenames.size().
Decide inside the loop whether to use configuration_basenames.at(0) or at(index).